src/app/modules/item/pages/item-edit/item-edit.component.ts

```diff
@@ -1,21 +1,72 @@
-import { Component, OnDestroy } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { CurrentContentService } from 'src/app/shared/services/current-content.service';
+import { ItemDataSource } from '../../services/item-datasource.service';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { Item } from '../../http-services/get-item-by-id.service';
+
+interface itemValues {
+  title: string
+}
+
+const defaultItemValues: itemValues = {
+  title: ''
+};
 
 @Component({
   selector: 'alg-item-edit',
   templateUrl: './item-edit.component.html',
   styleUrls: ['./item-edit.component.scss']
 })
-export class ItemEditComponent implements OnDestroy {
+export class ItemEditComponent implements OnInit, OnDestroy {
+  item: Item;
+  itemForm: FormGroup;
+  itemValues: itemValues;
+
+  itemSubscription: Subscription;
+
+  changedInput: { [input: string]: string } = {};
 
   constructor(
     private currentContent: CurrentContentService,
+    private itemDataSource: ItemDataSource,
+    private formBuilder: FormBuilder,
   ) {
     this.currentContent.editState.next('editing');
   }
 
+  ngOnInit() {
+    this.itemValues = defaultItemValues;
+    this.initForm();
+    this.getCurrentItem();
+  }
+
   ngOnDestroy() {
     this.currentContent.editState.next('non-editable');
+    this.itemSubscription.unsubscribe();
+  }
+
+  initForm(): void {
+    this.itemForm = this.formBuilder.group({
+      title: [defaultItemValues.title]
+    });
+  }
+
+  getCurrentItem(): void {
+    this.itemSubscription = this.itemDataSource.item$.subscribe(item => {
+      this.item = item;
+      this.itemValues.title = item.string.title ? item.string.title : '';
+      this.itemForm.patchValue({
+        title: item.string.title
+      });
+    });
+  }
+
+  onInputChange(name: string, value: string): void {
+    if (typeof (value) == 'object')
+      return;
+
+    this.changedInput[name] = value;
```

Comment:
Probably you would want to reset this variable when item changes.
This is the problem is creating more "state" in a class, you increase the risk of bug through inconsistency.

Reply:
removed